File: api/models/research.py
"""Kimi Swarm 股票调研模型"""

from typing import Literal

from pydantic import BaseModel, Field


class ResearchCreateRequest(BaseModel):
    ts_code: str = Field(min_length=1, max_length=80, description="A 股代码或名称，如 600519.SH、贵州茅台")


class ResearchTaskResponse(BaseModel):
    task_id: str
    ts_code: str
    status: Literal["queued", "running", "completed", "failed"]
    progress: int = Field(default=0, ge=0, le=100)
    message: str = ""
    report: str = ""
    error: str = ""
    created_at: str
    started_at: str = ""
    completed_at: str = ""
    engine: str = "Kimi Code CLI"
    mode: str = "Swarm"
    skill: str = "zettaranc-perspective"
    trace_id: str = ""
    trace_available: bool = False
    trace_schema_version: int = 1
    agent_count: int = 0
    expected_agent_count: int = 5
    partial_result: bool = False
    partial_reason: str = ""
    last_activity_at: str = ""
    target_type: str = "stock"


class ResearchTaskListResponse(BaseModel):
    tasks: list[ResearchTaskResponse] = Field(default_factory=list)


class ResearchTaskSummary(BaseModel):
    task_id: str
    ts_code: str
    status: Literal["queued", "running", "completed", "failed"]
    progress: int = Field(default=0, ge=0, le=100)
    message: str = ""
    error: str = ""
    created_at: str
    started_at: str = ""
    completed_at: str = ""
    agent_count: int = 0
    expected_agent_count: int = 5
    partial_result: bool = False
    target_type: str = "stock"
    has_report: bool = False
    report_excerpt: str = ""


class ResearchHistoryResponse(BaseModel):
    tasks: list[ResearchTaskSummary] = Field(default_factory=list)
    total: int = 0
    page: int = 1
    page_size: int = 20
    status_counts: dict[str, int] = Field(default_factory=dict)

File: frontend/src/api/research.ts
import api from './client';

export type ResearchStatus = 'queued' | 'running' | 'completed' | 'failed';

export interface ResearchTask {
  task_id: string;
  ts_code: string;
  status: ResearchStatus;
  progress: number;
  message: string;
  report: string;
  error: string;
  created_at: string;
  started_at: string;
  completed_at: string;
  engine: string;
  mode: string;
  skill: string;
  trace_id: string;
  trace_available: boolean;
  trace_schema_version: number;
  agent_count: number;
  expected_agent_count: number;
  partial_result: boolean;
  partial_reason: string;
  last_activity_at: string;
  target_type: 'stock' | 'theme';
}

export interface ResearchTaskSummary {
  task_id: string;
  ts_code: string;
  status: ResearchStatus;
  progress: number;
  message: string;
  error: string;
  created_at: string;
  started_at: string;
  completed_at: string;
  agent_count: number;
  expected_agent_count: number;
  partial_result: boolean;
  target_type: 'stock' | 'theme';
  has_report: boolean;
  report_excerpt: string;
}

export interface ResearchHistoryResponse {
  tasks: ResearchTaskSummary[];
  total: number;
  page: number;
  page_size: number;
  status_counts: Record<string, number>;
}

export async function createResearch(tsCode: string): Promise<ResearchTask> {
  const { data } = await api.post<ResearchTask>('/research/', { ts_code: tsCode });
  return data;
}

export async function fetchResearch(taskId: string): Promise<ResearchTask> {
  const { data } = await api.get<ResearchTask>(`/research/${taskId}`);
  return data;
}

export async function fetchResearchHistory(params: {
  page?: number;
  pageSize?: number;
  status?: ResearchStatus;
  keyword?: string;
}): Promise<ResearchHistoryResponse> {
  const { data } = await api.get<ResearchHistoryResponse>('/research/history', {
    params: {
      page: params.page || 1,
      page_size: params.pageSize || 20,
      status: params.status,
      keyword: params.keyword || undefined,
    },
  });
  return data;
}
